Software/Python/easysensors.py

```diff
@@ -1744,122 +1746,99 @@ class LineFollower(Sensor):
          line_follower.read_raw_sensors()
 
     .. warning::
-         This class requires the :py:mod:`line_sensor` library.
+         This class requires the :py:mod:`di_sensors` library. Check the :py:class:`di_sensors.easy_line_follower.EasyLineFollower`
+         class to find out more about its implementation.
 
     """
 
     def __init__(self, port="I2C", gpg=None, use_mutex=False):
         """
         Constructor for initalizing a :py:class:`~easysensors.LineFollower` object.
 
-        :param str port = "I2C": The port to which we have connected the `Line Follower`_ sensor.
+        :param str port = "I2C": The port to which we have connected the line follower sensor. Can also be ``"AD1"``/``"AD2"`` for the `Line Follower`_ sensor.
         :param easygopigo3.EasyGoPiGo3 gpg = None: The :py:class:`~easygopigo3.EasyGoPiGo3` object that we need for instantiating this object.
         :param bool use_mutex = False: When using multiple threads/processes that access the same resource/device, mutexes should be enabled.
-        :raises ImportError: If the :py:mod:`line_follower` module couldn't be found.
+        :raises ImportError: If the ``di_sensors`` library couldn't be found.
         :raises TypeError: If the ``gpg`` parameter is not a :py:class:`~easygopigo3.EasyGoPiGo3` object.
         :raises IOError: If the line follower is not responding.
 
-        The only value the ``port`` parameter can take is ``"I2C"``.
-        The I2C ports' location on the `GoPiGo3`_ robot can be seen in the following graphical representation: :ref:`hardware-ports-section`.
+        The only value the ``port`` parameter can take for the `Old Line Follower`_ is the ``"I2C"`` and for the new `Line Follower`_ it can also be ``"AD1"`` and ``"AD2"``.
+        The ports' location on the `GoPiGo3`_ robot can be seen in the following graphical representation: :ref:`hardware-ports-section`.
 
         """
         if is_line_follower_accessible is False:
-            raise ImportError("Line Follower library not found")
+            raise ImportError("di_sensors library not found")
 
         try:
             self.set_descriptor("Line Follower")
             Sensor.__init__(self, port, "INPUT", gpg, use_mutex)
-            if line_sensor.read_sensor() == [-1, -1, -1, -1, -1]:
-                raise IOError("Line Follower not responding")
-            self.white_line = line_sensor.get_white_line()
-            self.black_line = line_sensor.get_black_line()
-            self._calculate_threshold()
+            if port == "AD1" or port == "AD2":
+                self._lf = EasyLineFollower(port, module_id = 2)
```

Comment:
what's this module_id?   Can't we make it transparent to the user? Like I did for distance sensor?

Reply:
Well, it is - cause you don't have to do anything about it - it's all selected automatically. And `module_id=2` stands for the new/black line follower, which is the only one available on the GoPiGo3. You wouldn't want to let it choose the module automatically when you already know it has to be the new/black line follower. Inherently speaking, letting it do it automatically can be more prone to errors than selecting the module manually.

If you read the documentation for the `module_id` parameter, you'll see it represents 2 modes of operation: automatic and manual. Value `-1` stands for automatic, `1` for the old/red line follower and `2` for the new/black line follower.